internal/controllers/clusterextension_controller_test.go

```diff
@@ -229,22 +245,27 @@ func TestClusterExtensionChannelVersionExists(t *testing.T) {
 	require.Equal(t, metav1.ConditionTrue, cond.Status)
 	require.Equal(t, ocv1alpha1.ReasonSuccess, cond.Reason)
 	require.Equal(t, "resolved to \"quay.io/operatorhubio/prometheus@fake1.0.0\"", cond.Message)
-	cond = apimeta.FindStatusCondition(clusterExtension.Status.Conditions, ocv1alpha1.TypeInstalled)
-	require.NotNil(t, cond)
-	require.Equal(t, metav1.ConditionUnknown, cond.Status)
-	require.Equal(t, ocv1alpha1.ReasonInstallationStatusUnknown, cond.Reason)
-	require.Equal(t, "bundledeployment status is unknown", cond.Message)
-
-	t.Log("By fetching the bundled deployment")
-	bd := &rukpakv1alpha2.BundleDeployment{}
-	require.NoError(t, cl.Get(ctx, types.NamespacedName{Name: extKey.Name}, bd))
-	require.Equal(t, "core-rukpak-io-registry", bd.Spec.ProvisionerClassName)
-	require.Equal(t, installNamespace, bd.Spec.InstallNamespace)
-	require.Equal(t, rukpakv1alpha2.SourceTypeImage, bd.Spec.Source.Type)
-	require.NotNil(t, bd.Spec.Source.Image)
-	require.Equal(t, "quay.io/operatorhubio/prometheus@fake1.0.0", bd.Spec.Source.Image.Ref)
 
-	verifyInvariants(ctx, t, reconciler.Client, clusterExtension)
+	// we're skipping based on same loadError since Status.Condition setting path is skipped and bundle deployment is absent
+	// also invariants will be different
+	if err != nil && err.Error() != utilerrors.NewAggregate([]error{loadError}).Error() {
+		cond = apimeta.FindStatusCondition(clusterExtension.Status.Conditions, ocv1alpha1.TypeInstalled)
+		require.NotNil(t, cond)
+		require.Equal(t, metav1.ConditionUnknown, cond.Status)
+		require.Equal(t, ocv1alpha1.ReasonInstallationStatusUnknown, cond.Reason)
+		require.Equal(t, "bundledeployment status is unknown", cond.Message)
+
+		t.Log("By fetching the bundled deployment")
+		bd := &rukpakv1alpha2.BundleDeployment{}
+		require.NoError(t, cl.Get(ctx, types.NamespacedName{Name: extKey.Name}, bd))
+		require.Equal(t, "core-rukpak-io-registry", bd.Spec.ProvisionerClassName)
+		require.Equal(t, installNamespace, bd.Spec.InstallNamespace)
+		require.Equal(t, rukpakv1alpha2.SourceTypeImage, bd.Spec.Source.Type)
+		require.NotNil(t, bd.Spec.Source.Image)
+		require.Equal(t, "quay.io/operatorhubio/prometheus@fake1.0.0", bd.Spec.Source.Image.Ref)
+
+		verifyInvariants(ctx, t, reconciler.Client, clusterExtension)
```

Comment:
By returning an error on `Storage.Load()` intentionally we avoided a lot of mocking, but also code paths that would have set these exactly as expected. Also, the subsequent checking triggered here is kind of brittle.

Reply:
There's quite a bit of code path beyond `storage.Load` that we will need to cover pretty quickly. I don't think it needs to be covered _immediately_ in the initial PR that merges, but I think we'll want to beef up coverage for cases where `Load` returns non-errors.

Reply:
agree. also `verifyInvariants` is restored here.